src/include/splash/basetypes/ColTypeDim.hpp

```diff
@@ -51,6 +56,35 @@ namespace splash
         {
             return Dimensions::getSize();
         }
+
+        std::string toString() const
+        {
+            return "Dimension";
+        }
+
+        static CollectionType* genType(hid_t datatype_id)
+        {
+            bool found = false;
+            H5T_class_t h5_class = H5Tget_class(datatype_id);
+            if(h5_class == H5T_COMPOUND){
+                if(H5Tget_nmembers(datatype_id) == 3){
+                    if(H5Tget_size(datatype_id) == Dimensions::getSize()){
+                        char* m0 = H5Tget_member_name(datatype_id, 0);
+                        char* m1 = H5Tget_member_name(datatype_id, 1);
+                        char* m2 = H5Tget_member_name(datatype_id, 2);
+                        if( (strcmp("x", m0) || strcmp("y", m1) || strcmp("z", m2)) == 0)
```

Comment:
there is no reason to outsmart other programmers. use "==0" for all individual checks as this is what you mean